sensorhub-core/src/main/java/org/sensorhub/impl/comm/UDPCommProvider.java

```diff
@@ -38,83 +41,81 @@
  * @author Alex Robin
  * @since Dec 12, 2015
  */
-public class UDPCommProvider extends AbstractModule<UDPCommProviderConfig> implements ICommProvider<UDPCommProviderConfig>
-{
+public class UDPCommProvider extends AbstractModule<UDPCommProviderConfig> implements ICommProvider<UDPCommProviderConfig> {
     static final Logger log = LoggerFactory.getLogger(UDPCommProvider.class.getSimpleName());
-    
+    private final Set<Consumer<ConnectionEventArgs>> listeners = new HashSet<>();
```

Comment:
Same comment as in TCP Comm Provider regarding  need for or understanding need for more than one listener as well as choice of Set over ArrayList